src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -4313,6 +4313,341 @@ static void Main()
+                Diagnostic(ErrorCode.ERR_CallArgMixing, "local").WithArguments("R.Add(in int)", "x").WithLocation(24, 23),
+                // (31,16): warning CS9080: Use of variable 'r' in this context may expose referenced variables outside of their declaration scope
+                //         return r; // 3
+                Diagnostic(ErrorCode.WRN_EscapeVariable, "r").WithArguments("r").WithLocation(31, 16),
+                // (45,23): warning CS9091: This returns local 'local' by reference but it is not a ref local
+                //         r = new R() { local }; // 4
+                Diagnostic(ErrorCode.WRN_RefReturnLocal, "local").WithArguments("local").WithLocation(45, 23));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75802")]
+        public void CollectionInitializer_UnscopedRef_OtherScopes()
+        {
+            var source = """
+                using System.Collections;
+                using System.Diagnostics.CodeAnalysis;
+
+                class C
+                {
+                    R M1(ref int param)
+                    {
+                        var r = new R() { param };
+                        return r; // 1
+                    }
+
+                    R M2(ref int param)
+                    {
+                        var r = new R(param) { param };
+                        return r;
+                    }
+
+                    R M3([UnscopedRef] ref int param)
+                    {
+                        var r = new R() { param };
+                        return r;
+                    }
+
+                    R M4([UnscopedRef] ref int x, ref int y)
+                    {
+                        var r = new R(x) { y };
+                        return r; // 2
+                    }
+
+                    R M5([UnscopedRef] ref int x, ref int y)
+                    {
+                        var r = new R() { x, y };
+                        return r; // 3
+                    }
+                }
+
+                ref struct R : IEnumerable
+                {
+                    public R() { }
+                    public R(in int p) : this() { }
+
+                    public void Add([UnscopedRef] in int x) { }
+                    IEnumerator IEnumerable.GetEnumerator() => throw null;
+                }
+                """;
+
+            // The `new R() { param }` expression could be inferred to have ReturnOnly scope in M1 (then there would be no errors),
+            // however the ref safety analysis is currently more conservative and only infers either CallingMethod or local scope for the expression.
+            // It is equivalent to what would happen for `var r = new R(); r.Add(param);` where the scope of `r` is also either CallingMethod or local (if `scoped` keyword is used).
+            // Similarly `new R(param) { param }` in M2 is equivalent to `var r = new R(param); r.Add(param);` so the scope of `r` is ReturnOnly.
```

Comment:
Made a change to make it work as described in this comment. An offline discussion with @RikkiGibson led to me to add this test case and I *think* this is how it should behave?

Reply:
Talked with Rikki offline, I will think about a more robust approach. Basically now we are doing arg mixing check inside GetValEscape which is sort of a hack (we normally never to CheckValEscape inside GetValEscape) but it's simple to implement and probably can be considered good enough for this edge case scenario. The alternative approach would be to extract some sort of GetInvocationEscape utility that would analyze the invocation arguments and tell us what can escape into the receiver (now similar utilities tell us only what escapes into the return value). We could then hopefully use this also for similar bugs like https://github.com/dotnet/roslyn/issues/63306 and it wouldn't feel so hacky.